### PR TITLE
Desription for gaming sessions in battleship contract

### PR DESCRIPTION
**Pull request by LouiseMedova**
Updated sections:

-
-
-

### CODE REVIEW

File: docs/examples/Gaming/battleship.md [hunk 1]
```diff
@@ -324,6 +324,61 @@ extern fn state() {
...
 ```
+## Gaming sessions
+To further enhance the gaming experience and make it more user-friendly, the introduction of gaming sessions offers a significant improvement. Players now have the option to create a trusted game session, during which they are not required to sign every transaction individually. This trusted session creates a new temporary private key, valid for a specific time window, which empowers the temporary account to sign and send specific transactions on behalf of the player.
+
+These sessions allow users to establish predefined rules for interacting with a Dapp, offering the flexibility for unrestricted usage within these guidelines, eliminating the need to authorize each transaction separately.This approach not only facilitates a seamless Dapp experience but also ensures the security of assets, as users can specify the permissible actions for the Dapp.
+
+At the session's conclusion, whether it's the end of the indicated window or when the user decides to terminate it, the temporary key becomes obsolete and is discarded. 
+
+To initiate a session, a player sends a message to the contract (assuming the contract supports this message type):
+```rust
```
Comment (shamilsan) [inline docs/examples/Gaming/battleship.md:335]: Need to set a path to the file as we can check the difference with GitHub using this script: https://github.com/gear-foundation/gear-wiki/blob/master/script_checking_code.js
```suggestion
```rust title="battleship/src/contract.rs"
```

File: docs/examples/Gaming/battleship.md [hunk 2]
```diff
@@ -324,6 +324,61 @@ extern fn state() {
...
+CreateSession { 
+    key: ActorId, 
+    duration: u64, 
+    allowed_actions: Vec<ActionsForSession>
+    },
```
Comment (shamilsan) [inline docs/examples/Gaming/battleship.md:340]:
```suggestion
},
```

File: docs/examples/Gaming/battleship.md [hunk 3]
```diff
@@ -324,6 +324,61 @@ extern fn state() {
...
+```
+where:
+- `key` is the temporary account playing on the user's behalf;
+- `duration` is the time frame for the session's validity;
+- `allowed_actions` are the messages the temporary account can send.
+
+A session is structured as follows, allowing a player to set it up before starting the game:
+
+```rust
```
Comment (shamilsan) [inline docs/examples/Gaming/battleship.md:349]: Also,
```suggestion
```rust title="..."
```

File: docs/examples/Gaming/battleship.md [hunk 4]
```diff
@@ -324,6 +324,61 @@ extern fn state() {
...
+    pub allowed_actions: Vec<ActionsForSession>,
+}
+```
+
+After creating a session, the player must also fund the temporary account, which can be done via a voucher.
+
+In this example, the messages:
+```rust
+    StartGame { ships: Ships },
+    Turn { step: u8 },
```
Comment (shamilsan) [inline docs/examples/Gaming/battleship.md:365]:
```suggestion
StartGame { ships: Ships },
Turn { step: u8 },
```

File: docs/examples/Gaming/battleship.md [hunk 5]
```diff
@@ -324,6 +324,61 @@ extern fn state() {
     }
 }
 ```
+## Gaming sessions
+To further enhance the gaming experience and make it more user-friendly, the introduction of gaming sessions offers a significant improvement. Players now have the option to create a trusted game session, during which they are not required to sign every transaction individually. This trusted session creates a new temporary private key, valid for a specific time window, which empowers the temporary account to sign and send specific transactions on behalf of the player.
+
+These sessions allow users to establish predefined rules for interacting with a Dapp, offering the flexibility for unrestricted usage within these guidelines, eliminating the need to authorize each transaction separately.This approach not only facilitates a seamless Dapp experience but also ensures the security of assets, as users can specify the permissible actions for the Dapp.
```
Comment (shamilsan) [inline docs/examples/Gaming/battleship.md:330]:
```suggestion
These sessions allow users to establish predefined rules for interacting with a Dapp, offering the flexibility for unrestricted usage within these guidelines, eliminating the need to authorize each transaction separately. This approach not only facilitates a seamless Dapp experience but also ensures the security of assets, as users can specify the permissible actions for the Dapp.
```

**Review by shamilsan — APPROVED** [review]

**Review by AndrePanin — APPROVED** [review]

**Pull request merged**